ta4j-core/src/main/java/org/ta4j/core/Decimal.java

```diff
@@ -202,6 +202,84 @@ public Decimal sqrt() {
         return new Decimal(StrictMath.sqrt(delegate.doubleValue()));
     }
 
+    /**
+     * Returns the correctly rounded hyperbolic tangent of the <code>double</code> value of this {@code Decimal}.
+     * /!\ Warning! Uses the {@code StrictMath#tanh(double)} method under the hood.
+     * @return the hyperbolic tangent of {@code this}
+     * @see StrictMath#tanh(double)
+     */
+    public Decimal tanh() {
+        if (this == NaN) {
+            return NaN;
+        }
+        return new Decimal(StrictMath.tanh(delegate.doubleValue()));
+    }
+    
+    /**
+     * Returns the correctly rounded arc tangent of the <code>double</code> value of this {@code Decimal}.
+     * /!\ Warning! Uses the {@code StrictMath#atan(double)} method under the hood.
+     * @return the arc tangent of {@code this}
+     * @see StrictMath#atan(double)
+     */
+    public Decimal atan() {
+        if (this == NaN) {
+            return NaN;
+        }
+        return new Decimal(StrictMath.atan(delegate.doubleValue()));
+    }
+    
+    /**
+     * Returns the correctly rounded trigonometric cosine of an angle of the <code>double</code> value of this {@code Decimal}.
+     * /!\ Warning! Uses the {@code StrictMath#cos(double)} method under the hood.
+     * @return the trigonometric cosine of {@code this}
+     * @see StrictMath#cos(double)
+     */
+    public Decimal cos() {
+        if (this == NaN) {
+            return NaN;
+        }
+        return new Decimal(StrictMath.cos(delegate.doubleValue()));
+    }
+    
+    /**
+     * Returns the correctly rounded trigonometric sine of an angle of the <code>double</code> value of this {@code Decimal}.
+     * /!\ Warning! Uses the {@code StrictMath#sin(double)} method under the hood.
+     * @return the trigonometric sine of {@code this}
+     * @see StrictMath#sin(double)
+     */
+    public Decimal sin() {
+        if (this == NaN) {
+            return NaN;
+        }
+        return new Decimal(StrictMath.sin(delegate.doubleValue()));
+    }
+	
+    /**
+     * Returns the correctly rounded the hyperbolic cosine of the <code>double</code> value of this {@code Decimal}.
+     * /!\ Warning! Uses the {@code StrictMath#cosh(double)} method under the hood.
+     * @return the hyperbolic cosine of {@code this}
+     * @see StrictMath#cosh(double)
+     */
+    public Decimal cosh() {
+        if (this == NaN) {
+            return NaN;
+        }
+        return new Decimal(StrictMath.cosh(delegate.doubleValue()));
+    }
+    
+    /**
+     * Returns the correctly rounded hyperbolic sine of the <code>double</code> value of this {@code Decimal}.
+     * /!\ Warning! Uses the {@code StrictMath#sinh(double)} method under the hood.
+     * @return the hyperbolic sine of {@code this}
+     * @see StrictMath#sinh(double)
+     */
+    public Decimal sinh() {
+        if (this == NaN) {
+            return NaN;
+        }
+        return new Decimal(StrictMath.sinh(delegate.doubleValue()));
+    }
+
```

Comment:
So far `public Decimal sqrt()` is the only function that uses a _dirty_ double transformation... (used by 6 indicators at the moment). I would prefer not to implement this kind of math in the API of Decimal, but, if needed, in the corresponding indicator. Otherwise we support, like you said, nether accuracy nor performance

Reply:
You know my opinion. Sqrt/Exp and the like are common math and exist very long time in the world of mathematics. There is a reason for such formulas. It would be pitty not to have common math within this library. As u see, it takes the whole story of "accuracy" ad absurdum. You cannot guarantee this for indicators, and you really dont need to! If a indicator needs that tool, high accuracy is - at least there - no more satisifed. And I dont talk about all the other indicators which inherits this "unaccuracy" indicator within its formula. As u said - 6 indicators make direct use of them - and indirect maybe a lot more. This is only the beginning..there wil come more who wants to have common math within its formula and all their indicators have a loos of precision with the loos of performance. Please understand, you dont need such accuracy within indicators. I cannot find any use case where it matters. Think practically. I only want to help to have a better piece of software. Please dont think, I am against your opinion, has nothing to do with you. Your work is great and I like your effort. Thanks for that.